pkg/plugins/policies/meshtrace/api/v1alpha1/validator.go

```diff
@@ -0,0 +1,120 @@
+package v1alpha1
+
+import (
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/asaskevich/govalidator"
+	"golang.org/x/exp/slices"
+
+	common_proto "github.com/kumahq/kuma/api/common/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core/validators"
+	matcher_validators "github.com/kumahq/kuma/pkg/plugins/policies/matchers/validators"
+	"github.com/kumahq/kuma/pkg/util/validation"
+)
+
+func (r *MeshTraceResource) validate() error {
+	var verr validators.ValidationError
+	path := validators.RootedAt("spec")
+	verr.AddErrorAt(path.Field("targetRef"), validateTop(r.Spec.GetTargetRef()))
+	verr.AddErrorAt(path.Field("default"), validateDefault(r.Spec.GetDefault()))
+	return verr.OrNil()
+}
+func validateTop(targetRef *common_proto.TargetRef) validators.ValidationError {
+	targetRefErr := matcher_validators.ValidateTargetRef(targetRef, &matcher_validators.ValidateTargetRefOpts{
+		SupportedKinds: []common_proto.TargetRef_Kind{
+			common_proto.TargetRef_Mesh,
+			common_proto.TargetRef_MeshSubset,
+			common_proto.TargetRef_MeshService,
+			common_proto.TargetRef_MeshServiceSubset,
+			common_proto.TargetRef_MeshGatewayRoute,
+		},
+	})
+	return targetRefErr
+}
+
+func validateDefault(conf *MeshTrace_Conf) validators.ValidationError {
+	var verr validators.ValidationError
+
+	if conf == nil {
+		verr.AddViolation("", validation.MustBeDefined())
+		return verr
+	}
+
+	if len(conf.GetBackends()) != 1 {
+		verr.AddViolation("backends", "must have one backend defined")
+	} else {
+		backend := conf.GetBackends()[0]
```

Comment:
how about
```
usedBackends := 0
if backend.GetDatadog() != nil {
  usedBackends += 1
}
if backend.GetZipkin() != nil {
  usedBackend += 1 
}
if usedBackends != 1 {
...
```
or
```
if (backend.GetDatadog() != nil) != (backend.GetZipkin() != nil) { // xor
  ...
}
```

and we drop the `Bool2Int` util

Reply:
We actually want a **negative** xor so it will be 

```
(backend.GetDatadog() != nil) == (backend.GetZipkin() != nil)
```

We want to add error if both of them are defined (both are false) or both are not defined (both are true).